### PR TITLE
Fix verification cancellation

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Changes
When the LSP server is notified of a change, cancel all verification tasks of the current document version.

### Testing
Added tests

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyLanguageServer/Workspace/CompilationManager.cs [hunk 1]
```diff
@@ -114,6 +113,10 @@ public class CompilationManager {
       var resolvedDocument = await ResolvedDocument;
       var translatedDocument = await PrepareVerificationTasksAsync(resolvedDocument, cancellationSource.Token);
       documentUpdates.OnNext(translatedDocument);
+      foreach (var task in translatedDocument.VerificationTasks!) {
+        cancellationSource.Token.Register(task.Cancel);
```
Comment (MikaelMayer) [inline Source/DafnyLanguageServer/Workspace/CompilationManager.cs:117]: Oh !! Is this where you are resolving the fact that tasks were not cancelled?

File: Source/DafnyLanguageServer.Test/Various/CancelVerificationTest.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+using Microsoft.Dafny.LanguageServer.Workspace;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Various {
+  [TestClass]
+  public class CancelVerificationTest : ClientBasedLanguageServerTest {
+
+    [TestMethod]
+    public async Task ChangingTheDocumentStopsOnChangeVerification() {
```
Comment (MikaelMayer) [inline Source/DafnyLanguageServer.Test/Various/CancelVerificationTest.cs:18]: Fantastic chess move to verify this behavior for both the onchange and onsave !

**Review by MikaelMayer — APPROVED** [review]
I feel so relieved that cancellation is fixed. Thanks

File: Source/DafnyLanguageServer/Workspace/CompilationManager.cs (hunk 1 above)
Comment (keyboardDrummer) [reply Source/DafnyLanguageServer/Workspace/CompilationManager.cs:117]: Jep, pretty simple.

**Pull request merged**